gc/src/trace.rs

```diff
@@ -344,11 +417,13 @@ unsafe impl<T: Trace> Trace for LinkedList<T> {
         }
     });
 }
+unsafe impl<T: EmptyTrace> EmptyTrace for LinkedList<T> {}
 
 impl<T: ?Sized> Finalize for PhantomData<T> {}
 unsafe impl<T: ?Sized> Trace for PhantomData<T> {
     unsafe_empty_trace!();
 }
+unsafe impl<T: EmptyTrace> EmptyTrace for PhantomData<T> {}
```

Comment:
```suggestion
unsafe impl<T> EmptyTrace for PhantomData<T> {}
```

By definition, `PhantomData<T>` does not contain a `T`.